app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -62,6 +62,23 @@ def with_fallback_urls(urls)
     false
   end
 
+  # TODO: Don't ignore idle timeout error when following bug in qpid_proton is fixed:
+  # https://bugzilla.redhat.com/show_bug.cgi?id=1418026
+  # Until then, we don't want this error to keep restarting our event catcher in case there are no events emitted
+  # for 16 seconds.
+  def run_with_idle_timeout_ignored
+    loop do
+      begin
+        connection.run
+      rescue MiqException::MiqHostError => e
+        raise unless e.message.include?('local-idle-timeout expired')
```

Comment:
I presume there is no "error code" or something similar that you could check for?